demo/c++/build.py

```diff
@@ -36,14 +36,15 @@
 demo_env['CXXFLAGS'] = copy(env['LIBMAPNIK_CXXFLAGS'])
 demo_env.Append(CPPDEFINES = env['LIBMAPNIK_DEFINES'])
 
+demo_env['LINKFLAGS'] = '-l%s ' % env['MAPNIK_NAME'] +  demo_env['LINKFLAGS']
```

Comment:
Ensure `-lmapnik` appears before `libc++.a` so all std:: symbols resolved. 
NOTE: @springmeyer mentioned that [lld](https://lld.llvm.org/) can resolve symbols in any order (needs checking to see if this is a future solution)